docs/reference/api/drivers/SAI.md

```diff
@@ -0,0 +1,52 @@
+## Serial Audio Interface (SAI)
+
+The Serial Audio Interface (SAI) provides a class to communicate with audio devices. SAI is a broader feature that encompasses audio interfaces such as I2S.
+
+SAI uses a four wire serial protocol consisting of:
+  * Master Clock Pin  (mclk)
+  * Data Pin          (sd)
+  * Bit Clock pin     (bclk)
+  * Word Clock Pin    (wclk)
+
+The master clock is used to optionally synchronize the device's IO lines. When using SAI to communicate in loopback or with an on-target audio codec there may be an internally shared master clock and this pin is not required. The `mclk_source` field in the format object that may be optionally passed into the SAI constructor can be set to internal, external or sibling.
+
+The bit clock fires on each bit of data transmitted across the data lines and its frequency is determined by the sampling rate, number of bits per channel, and number of channels. The sample rate is highly device dependent and needs to be configured for each target (typically found in the target's device.h file).
```

Comment:
A good amount of this section may be better suited for the contributing docs page (pending). Thoughts?